ooiservices/app/uframe/controller.py

```diff
@@ -87,46 +81,45 @@ def split_stream_name(ui_stream_name):
     all_of_it = '_'.join([first_part, stream_type, stream])
     return all_of_it
 
-
 def iso_to_timestamp(iso8601):
     dt = parse_date(iso8601)
-    t = (dt - datetime(1970, 1, 1, tzinfo=pytz.utc)).total_seconds()
+    t = (dt - datetime(1970,1,1,tzinfo=pytz.utc)).total_seconds()
     return t
 
-
 def dict_from_stream(mooring, platform, instrument, stream_type, stream):
     from ooiservices.app.main.routes import get_display_name_by_rd
     HOST = str(current_app.config['HOST'])
     PORT = str(current_app.config['PORT'])
     SERVICE_LOCATION = 'http://'+HOST+":"+PORT
-    response = get_uframe_stream_contents(mooring, platform, instrument, stream_type, stream)
+    ref = mooring + "-" + platform + "-" + instrument
+    response = get_uframe_stream_metadata_times(ref)
     stream_name = '_'.join([stream_type, stream])
     ref = '-'.join([mooring, platform, instrument])
     if response.status_code != 200:
         raise IOError("Failed to get stream contents from uFrame")
     data = response.json()
-    data_dict = {}
-    preferred = data[0][u'preferred_timestamp']
-    data_dict['start'] = data[0]['pk']['time'] - COSMO_CONSTANT
-    data_dict['end'] = data[-1]['pk']['time'] - COSMO_CONSTANT
-    data_dict['reference_designator'] = '-'.join([mooring, platform, instrument])
+    data_dict = {}   
+    data_dict['start'] = data[0]['beginTime']
+    data_dict['end'] = data[0]['endTime']
+    data_dict['reference_designator'] = data[0]['sensor']
     data_dict['display_name'] = get_display_name_by_rd(ref)
-    data_dict['csv_download'] = "/".join([SERVICE_LOCATION, 'uframe/get_csv', stream_name, ref])
-    data_dict['json_download'] = "/".join([SERVICE_LOCATION, 'uframe/get_json', stream_name, ref])
-    data_dict['netcdf_download'] = "/".join([SERVICE_LOCATION, 'uframe/get_netcdf', stream_name, ref])
-    data_dict['profile_json_download'] = "/".join([SERVICE_LOCATION, 'uframe/get_profiles', ref, stream_name])
+    data_dict['download'] = {"csv":"/".join([SERVICE_LOCATION, 'uframe/get_csv', stream_name, ref]),
+                             "json":"/".join([SERVICE_LOCATION, 'uframe/get_json', stream_name, ref]),
+                             "netcdf":"/".join([SERVICE_LOCATION, 'uframe/get_netcdf', stream_name, ref]),
+                             "profile":"/".join([SERVICE_LOCATION, 'uframe/get_profiles', stream_name, ref])
+                            }
+    #data_dict['csv_download'] = "/".join([SERVICE_LOCATION, 'uframe/get_csv', stream_name, ref])
+    #data_dict['json_download'] = "/".join([SERVICE_LOCATION,'uframe/get_json',stream_name,ref])
+    #data_dict['netcdf_download'] = "/".join([SERVICE_LOCATION,'uframe/get_netcdf',stream_name,ref])
+    #data_dict['profile_json_download'] = "/".join([SERVICE_LOCATION,'uframe/get_profiles',ref,stream_name])
     data_dict['stream_name'] = stream_name
-    data_dict['variables'] = data[1].keys()
-    data_dict['variable_types'] = {k : type(data[1][k]).__name__ for k in data[1].keys() }
```

Comment:
Definitely.  We have to come up wiht a work around because this is no longer availbalbe here... Maybe request one datapoint worth of data, including L1/L2?